packages/vscode-extension/src/devices/preview.ts

```diff
@@ -6,22 +6,69 @@ import { Logger } from "../Logger";
 import { Platform } from "../utilities/platform";
 import { RecordingData, TouchPoint } from "../common/Project";
 
-interface ReplayPromiseHandlers {
+type VideoRecordingPromiseType = "recording" | "replay";
+
+interface VideoRecordingPromiseHandlers {
   resolve: (value: RecordingData) => void;
   reject: (reason?: any) => void;
 }
 
 export class Preview implements Disposable {
+  private videoRecordingPromises: Map<
+    VideoRecordingPromiseType,
+    VideoRecordingPromiseHandlers | undefined
+  >;
   private subprocess?: ChildProcess;
   public streamURL?: string;
-  private lastReplayPromise?: ReplayPromiseHandlers;
 
-  constructor(private args: string[]) {}
+  constructor(private args: string[]) {
+    this.videoRecordingPromises = new Map<
+      VideoRecordingPromiseType,
+      VideoRecordingPromiseHandlers | undefined
+    >();
+
+    this.videoRecordingPromises.set("recording", undefined);
+    this.videoRecordingPromises.set("replay", undefined);
```

Comment:
we don't need undefined value there. If its not there `map.get(key)` will return undefined